src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -421,7 +428,7 @@ protected function build(ContainerBuilder $container)
      */
     protected function getContainerClass()
     {
-        return $this->name.ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
+        return 'app'.md5(\get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
```

Comment:
I don't see why we need to hash the class as it makes the container class ugly and does not provide much value.
I would replace `'app'.md5(\get_class($this)).` with `str_replace('\', '', \get_class($this)).` This way it's a valid class name as well and more readable. Now that we are at it we could also simplify the suffix 'ProjectContainer' to 'Container'

Reply:
Doing so bears the small risk of class name collisions (for example `App\Kernel` and `AppKernel` would end with the same generated class name). But we could replace backslashes with underscores if we want a more readable class name.

Reply:
Done